client/engine/metrics.go

```diff
@@ -90,23 +78,32 @@ func NewMetricsRecorder(me types.Address, metrics MetricsApi) *MetricsRecorder {
 	}
 }
 
-// RecordDuration records the duration of the given function for the metric specified by name
-func (o *MetricsRecorder) RecordDuration(name string, funcToTime func()) {
+// RecordMessageReceived records the duration of the function
+// It should be called at the start of the function like so  `defer e.metrics.RecordFunctionDuration()()`
+func (o *MetricsRecorder) RecordFunctionDuration() func() {
```

Comment:
Is it worth it to write a unit test for this function?

Reply:
I think a test would be a good idea but I'm going to defer it to get this PR merged in.